### PR TITLE
Implemented Atmos Hud

### PR DESCRIPTION
**Pull request by Setheri**
-3 modes, Temperature, Pressure, Off
-Overlays colors to indicate relative difference from the "normal" station temp and pressure
-Blue is low temp, red is high temp
-Green is low pressure, yellow is high pressure
-Action button in top left corner makes toggling modes simpler (no removing the HUD to change it)
-Only relevant (Not unsimulated, space, or dense) turfs are colored, and only the turf is colored by the HUD
-Custom yellow icons by Lichling(TM)

Example image: http://i.gyazo.com/846d9ec87a57ed5e3f50c7c203c3de40.png

### CODE REVIEW

File: code/modules/clothing/glasses/hud.dm [hunk 1]
```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
...
+	name = "Atmospheric Scanner HUD"
+	desc = "An advanced display used to observe atmospheric data in an area."
+	icon_state = "atmoshud"
+	item_state = "atmoshud"
+	action_button_name = "Set HUD Mode"
+	var/intensity = 64 // The value, 0-255, that determines opaqueness of the overlay
+	var/list/images = list()
+	var/list/modes = list("Pressure","Temperature","Off")
+	var/currentmode = 1 // Represents which mode in the list modes is currently active
+	var/const/basetemp = 293.15
```
Comment (xXTheDracheXx) [inline code/modules/clothing/glasses/hud.dm]: Replace the constants with the defines in math.dm

File: code/modules/clothing/glasses/hud.dm [hunk 2]
```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
...
+	var/const/basepressure = 101.325
+
+/obj/item/clothing/glasses/hud/atmos/proc/select_mode()
+	if (!modes.len) return //No possible modes? Must be broked
+	if (currentmode < 1) //Just Checking to prevent problems
```
Comment (xXTheDracheXx) [inline code/modules/clothing/glasses/hud.dm]: Here's for trying something: if(currentmod % mods.len == 0) currentmode = 1

File: code/modules/clothing/glasses/hud.dm [hunk 3]
```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
...
+		currentmode = 1
+	else currentmode++
+	if (currentmode > modes.len)
+		currentmode = 1
+
+	for (var/turf/T in images) //After switching modes, erase the images we have.
```
Comment (xXTheDracheXx) [inline code/modules/clothing/glasses/hud.dm]: Shouldn't it be var/image/I in images? Is the images list filled with turfs?

File: code/modules/clothing/glasses/hud.dm [hunk 4]
```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
...
+	add_fingerprint(user)
+	..()
+
+/obj/item/clothing/glasses/hud/atmos/equipped(mob/M, slot)
+	if (slot != slot_glasses)
+		for (var/turf/I in src.images)
+			janitor(I)
+	..()
+
+/obj/item/clothing/glasses/hud/atmos/proc/janitor(var/turf/T)
```
Comment (xXTheDracheXx) [inline code/modules/clothing/glasses/hud.dm]: You want the function name to best describe what it does. janitor() is not very descriptive.

File: code/modules/clothing/glasses/hud.dm [hunk 5]
```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
...
+
+		else //Make the new image
+
+			holder = new/image()
+			holder.icon = 'icons/effects/effects.dmi'
+			holder.icon_state = "overlay"
+			holder.loc = T
+			holder.layer = T.layer
+
+			if(modes[currentmode] == "Temperature") //MATH!
```
Comment (xXTheDracheXx) [inline code/modules/clothing/glasses/hud.dm]: As far as I can see, the next few lines is an exact copy of the code above. It'd be better if you put this in a function.

File: code/modules/clothing/glasses/hud.dm (hunk 3 above)
Comment (Setheri) [inline code/modules/clothing/glasses/hud.dm]: Images[turf] = the image used for that turf.

File: code/modules/clothing/glasses/hud.dm (hunk 4 above)
Comment (FlavoredCactus) [inline code/modules/clothing/glasses/hud.dm]: wzhzhzh()

**Pull request merged**